src/PrestaShopBundle/Form/Exception/DataProviderError.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Exception;
+
+class DataProviderError
```

Comment:
Maybe we can improve the naming?

Here it's called `DataProviderError` because it's "an error returned by the DataProvider". But tomorrow we could decide to return this from another class / kind of class so the naming would become wrong.

Here are some suggestions
- ConfigurationFormValidationError => because it's errors found while validating the data input inside a configuration form
- InvalidConfigurationDataError
- WrongConfigurationDataError
- ConfigurationDataValidationError

Reply:
Thanks for multiple suggestions! Yeah what you are saying makes sense.

I like InvalidConfigurationDataError, it says more what went wrong than where it went wrong thus making it more universal. So implemented that.